src/Bedrock.Framework.Experimental/Protocols/Tls/BulkCiphers/BulkCipherProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Tls.BulkCiphers
+{
+    internal abstract class BulkCipherProvider
+    {
+        public abstract BulkCipher GetCipher<T>(BulkCipherType cipherType, Memory<byte> keyStorage) where T : BulkCipher, new();
```

Comment:
Why is this memory writable?

Reply:
Originally this was "OwnedMemory" because it came from a pool, however the BulkCipher needed to hold it for it's lifetime and then say when it was done. Is there still a concept of this now?